src/js/components/AppListComponent.jsx

```diff
@@ -271,60 +314,61 @@ var AppListComponent = React.createClass({
       });
     }
 
-    return nodesSequence;
+    return items;
   },
 
-  getGroupedNodes: function (apps, filterCounts) {
-    var currentGroup = this.props.currentGroup;
-
-    var appsInGroup = apps.filter(app => app.id.startsWith(currentGroup));
+  groupItems: function (items, currentGroup, filterCounts) {
+    //  Exclude items that are not in the current group
+    var itemsInCurrentGroup = items.filter(
+      item => item.id.startsWith(currentGroup)
+    );
 
-    appsInGroup.forEach(app => {
-      filterCounts.appsStatusesCount[app.status]++;
-      if (app.container != null && app.container.volumes != null) {
+    itemsInCurrentGroup.forEach(item => {
+      if (item.isGroup) {
+        return;
+      }
+      filterCounts.appsStatusesCount[item.status]++;
+      if (item.container != null && item.container.volumes != null) {
         filterCounts.appsVolumesCount++;
       }
-      app.health.forEach(health => {
+      item.health.forEach(health => {
         if (health.quantity) {
           filterCounts.appsHealthCount[health.state]++;
         }
       });
     });
 
-    return appsInGroup
-      .reduce((memo, app) => {
-        let relativePath = app.id.substring(currentGroup.length);
-        let pathParts = relativePath.split("/");
-        let isGroup = pathParts.length > 1;
+    // Exclude items that are in a sub group
+    itemsInCurrentGroup = itemsInCurrentGroup.filter(item => {
+      let relativePath = item.id.substring(currentGroup.length);
+      let pathParts = relativePath.split("/");
```

Comment:
`let`/`var` again (the rule is that `var` is preferred, and `let` should only be used where it clarifies block scope)